system/API/ResponseTrait.php

```diff
@@ -358,6 +358,9 @@ protected function format($data = null)
 			$this->formatter = $config->getFormatter($format);
 		}
 
+		// Recursively convert objects into associative arrays
+		$data = json_decode(json_encode($data), true);
```

Comment:
var_dump($data);exit;
```
/home/username/CodeIgniter4/system/API/ResponseTrait.php:361:
object(App\Entities\User)[44]
  public 'id' => int 1
  public 'email' => string 'username@gmail.com' (length=21)
  public 'firstname' => string 'John' (length=28)
  public 'lastname' => string 'Doe' (length=6)
  public 'gender' => string 'm' (length=1)
  public 'birthday' => string '1990-01-01' (length=10)
  public 'picture_url' => null
  public 'links' => 
    object(stdClass)[43]
      public 'site' => string 'https://username.github.io' (length=29)
      public 'github' => string 'https://github.com/username' (length=30)
      public 'twitter' => string 'https://twitter.com/username' (length=31)
      public 'facebook' => string 'https://facebook.com/username' (length=32)
  public 'active' => string 'y' (length=1)
  public 'online' => string 'n' (length=1)
  public 'is_candidate' => string 'n' (length=1)
  public 'created_at' => string '2017-08-03T19:54:45+00:00' (length=25)
  public 'updated_at' => string '2017-08-04T01:21:49+00:00' (length=25)
  protected 'deleted' => string '0' (length=1)
  public 'url' => string 'http://localhost:8080/users/1' (length=29)
  protected 'datamap' => 
    array (size=0)
      empty

```

The objects must be recursively converted to array... Otherwise the XML formatter fail, no problem with JSON.